TELCODOCS-2127: Adding kernalpagesize API to PP (#87727)
openshift/openshift-docs: modules/configuring-kernal-page-size.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/low_latency_tuning/cnf-tuning-low-latency-nodes-with-perf-profile.adoc

:_mod-docs-content-type: PROCEDURE
[id="cnf-page-size-optimization_{context}"]
= Configuring kernel page sizes

Use the `kernelPageSize` specification in a performance profile to configure the kernel page size on a specific node. Specify larger kernel page sizes for memory-intensive, high-performance workloads. 

[NOTE]
====
For nodes with an x86_64 or AMD64 architecture, you can only specify `4k` for the `kernelPageSize` specification. For nodes with an AArch64 architecture, you can specify `4k` or `64k` for the `kernelPageSize` specification. The default value is `4k`.
====

.Prerequisites

> * Access to the cluster as a user with the `cluster-admin` role.
> 
> * Install the {oc-first}.

Reviewer: ```suggestion
* You have access to the cluster as a user with the `cluster-admin` role.

* You have installed the {oc-first}.
```

.Procedure

. Create a performance profile to target nodes where you want to configure the kernel page size by creating a YAML file that defines the `PerformanceProfile` resource:
+
.Example `pp-kernel-pages.yaml` file
[source,yaml]
----
apiVersion: performance.openshift.io/v1
kind: PerformanceProfile
metadata:
    name: example-performance-profile
#...
spec:
    kernelPageSize: "64k" <1>
    nodeSelector:
        node-role.kubernetes.io/worker: "" <2>
----
<1> This example specifies a kernel page size of `64k`. You can only specify `64k` for nodes with an AArch64 architecture. The default value is `4k`.
<2> This example targets nodes with the `worker` role.

> . Apply the performance profile to the cluster:

Reviewer: ```suggestion
. Apply the performance profile to the cluster by running the following command:
```

// Module included in the following assemblies:
//
// * scalability_and_performance/low_latency_tuning/cnf-tuning-low-latency-nodes-with-perf-profile.adoc

:_mod-docs-content-type: PROCEDURE
[id="cnf-page-size-optimization_{context}"]
= Configuring kernel page sizes

Use the `kernelPageSize` specification in a performance profile to configure the kernel page size on a specific node. Specify larger kernel page sizes for memory-intensive, high-performance workloads. 

[NOTE]
====
For nodes with an x86_64 or AMD64 architecture, you can only specify `4k` for the `kernelPageSize` specification. For nodes with an AArch64 architecture, you can specify `4k` or `64k` for the `kernelPageSize` specification. The default value is `4k`.
====

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

> * Install the OpenShift CLI (`oc`).

Reviewer 2: Use attribute:
```suggestion
* Install the {oc-first}.
```

.Procedure

> . Create a performance profile to target nodes where you want to configure the kernel page size:
> 
> .. Create a YAML file that defines the `PerformanceProfile` resource:

Reviewer 2: Feels a little funny to have a single-step procedure (which makes me wonder if technically this should be a `*`?) with two substeps.

Could this just be flattened out to two normal steps? Makes the sentence kinda long, but something like:
```suggestion
. Create a performance profile to target nodes where you want to configure the kernel page size by creating a YAML file that defines the `PerformanceProfile` resource:
```

+
.Example `pp-kernel-pages.yaml` file
[source,yaml]
----
apiVersion: performance.openshift.io/v1
kind: PerformanceProfile
metadata:
    name: example-performance-profile
#...
spec:
    kernelPageSize: "64k" <1>
    nodeSelector:
        node-role.kubernetes.io/worker: "" <2>
----
<1> This example specifies a kernel page size of `64k`. You can only specify `64k` for nodes with an AArch64 architecture. The default value is `4k`.
<2> This example targets nodes with the `worker` role.

> .. Apply the performance profile to the cluster:

Reviewer 2: And if the previous suggestion is taken, then this becomes:
```suggestion
. Apply the performance profile to the cluster:
```

+
[source,bash]
----
$ oc create -f pp-kernel-pages.yaml
----
+
.Example output
----
performanceprofile.performance.openshift.io/example-performance-profile created
----

.Verification

. Start a debug session on the node where you applied the performance profile by running the following command:
+
[source,bash]
----

> oc debug node/<node_name> <1>

Reviewer 2: Missing prompt:
```suggestion
$ oc debug node/<node_name> <1>
```
